drivers/timer/cortex_m_systick.c

```diff
@@ -421,13 +480,16 @@ void sys_clock_idle_exit(void)
 #endif /* CONFIG_CORTEX_M_SYSTICK_IDLE_TIMER */
 
 	if (last_load == TIMER_STOPPED) {
-		SysTick->CTRL |= SysTick_CTRL_ENABLE_Msk;
+		sys_write32(SYST_CSR_RUN, SYST_CSR);
```

Comment:
If I understand correctly, re-enabling systick will cause the counter to resume and on the next systick isr, `elapsed` calculation will be boggus if it falls in any of the case where `last_load` is used.

So, shouldn’t `last_load` also be set to a sound/meaningful value?
Or shouldn’t `elapsed` be fixed to account for the cases where `last_load` represents something else?

Reply:
Yeah, it's looks like a real major bug. IMHO it's better to stop using the "TIMER_STOPPED" at all. But it will be out of scope this PR.